subsys/bluetooth/mesh/prov.c

```diff
@@ -58,8 +59,8 @@ int bt_mesh_prov_reset_state(void (*func)(const uint8_t key[64]))
 		bt_mesh_attention(NULL, 0);
 	}
 
-	memset(&bt_mesh_prov_link.dhkey, 0,
-	       sizeof(bt_mesh_prov_link) - offsetof(struct bt_mesh_prov_link, dhkey));
+	(void)memset((char *)&bt_mesh_prov_link + offset, 0,
```

Comment:
Could you please cast to (uint8_t *) here instead of `char *`? `char` is technically of undefined width.

Reply:
Done.